gooddata-sdk/gooddata_sdk/compute/model/execution.py

```diff
@@ -54,15 +80,50 @@ def is_one_dim(self) -> bool:
     def is_two_dim(self) -> bool:
         return len(self.dimensions) == 2
 
-    def as_api_model(self) -> models.AfmExecution:
+    def _create_dimensions(self) -> list[models.Dimension]:
         dimensions = []
-
         for idx, dim in enumerate(self._dimensions):
             dimensions.append(models.Dimension(local_identifier=f"dim_{idx}", item_identifiers=dim))
 
-        execution = compute_model_to_api_model(attributes=self.attributes, metrics=self.metrics, filters=self.filters)
+        return dimensions
+
+    def _create_totals(self) -> Optional[list[models.Total]]:
+        if not self._totals:
+            return None
+
+        totals = []
+        for total in self._totals:
+            total_dims = [
+                models.TotalDimension(
+                    dimension_identifier=f"dim_{total_dim.idx}", total_dimension_items=total_dim.items
+                )
+                for total_dim in total.total_dims
+            ]
+
+            totals.append(
+                models.Total(
+                    local_identifier=total.local_id,
+                    metric=total.metric_local_id,
+                    function=total.aggregation.upper(),
+                    total_dimensions=total_dims,
+                )
+            )
+
+        return totals
 
-        result_spec = models.ResultSpec(dimensions=dimensions)
+    def _create_result_spec(self) -> models.ResultSpec:
+        dimensions = self._create_dimensions()
+        totals = self._create_totals()
+
+        if totals is None:
+            # creating model with totals=None bombs, even if the totals is optional in the definition
```

Comment:
How exactly does it bomb? Is there an error while ResultSpec initialization?

Reply:
The comment is confusing. `totals` can be given to ResultSpec but not as `None` but empty list.
As totals are optional it doesn't have to be there at all.
I'll remove the comment.
// Other option is to start sending empty totals if not filled but that's imo not necessary.